File: backend-java/src/main/java/com/oceanview/reservation/models/Offer.java
package com.oceanview.reservation.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;

@Data
@Document(collection = "offers")
public class Offer {
    @Id
    private String id;
    private String title;
    private String description;
    private String discountCode;
    private Boolean isActive = true;
    private java.util.Date createdAt = new java.util.Date();
}

File: backend/src/models/Offer.js
const mongoose = require("mongoose");

const offerSchema = new mongoose.Schema({
    title: { type: String, required: true },
    description: { type: String, required: true },
    discountCode: { type: String, required: false }, // e.g. "SUMMER24", optional
    isActive: { type: Boolean, default: true },
    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model("Offer", offerSchema);
